core/src/main/java/org/apache/calcite/rel/rules/JoinToCorrelateRule.java

```diff
@@ -77,18 +76,19 @@
    */
   public static final JoinToCorrelateRule JOIN =
       new JoinToCorrelateRule(LogicalJoin.class, RelFactories.LOGICAL_BUILDER,
-              "JoinToCorrelateRule", join -> SemiJoinType.of(join.getJoinType()));
+              "JoinToCorrelateRule", Join::getJoinType);
 
-  @Deprecated // to be removed (should use JOIN instead), kept for backwards compatibility
+  @Deprecated // To be removed (should use JOIN instead), kept for backwards compatibility
```

Comment:
After the current refactoring it seems that there is no need to have multiple instances of the rule for different kind of joins. Maybe a single instance of the rule can treat all kinds of joins. If that's the case we could remove deprecation from `INSTANCE` and rather put it on `JOIN`.

Reply:
I agree, will deprecate JOIN instead of INSTANCE